pmlc/dialect/pxa/transforms/vectorize.cc

```diff
@@ -66,6 +95,17 @@ class Impl {
 
           auto it = strideInfo->strides.find(index);
           if (it == strideInfo->strides.end()) {
+            // vector::ReductionOp doesn't support pxa's assign reduction.
+            // Also, make sure we handle only the supported types -
+            // see the vector::ReductionOp verification code
+            // (https://github.com/llvm/llvm-project/blob/master/mlir/lib/Dialect/Vector/VectorOps.cpp#L134).
+            Type eltType = op.getMemRefType().getElementType();
+            if (op.agg() == AtomicRMWKind::assign ||
+                (!eltType.isF32() && !eltType.isF64() &&
+                 !eltType.isSignlessInteger(32) &&
+                 !eltType.isSignlessInteger(64))) {
+              return failure();
```

Comment:
Use `op.emitError()` here.

Reply:
Adding a remark emission, as we discussed. The vectorization just doesn't happen if the vector::ReductionOperation doesn't support the type of the operand or the reduction type.